### PR TITLE
Add generic webhook payload to versioned types

### PR DESCRIPTION
**Pull request by csrwng**

### CODE REVIEW

File: pkg/build/webhook/generic/generic.go [hunk 1]
```diff
@@ -52,7 +42,7 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 		if len(body) == 0 {
 			return revision, true, nil
 		}
-		var data webHookEvent
+		var data api.GenericWebHookEvent
 		if err = json.Unmarshal(body, &data); err != nil {
```
Comment (smarterclayton) [inline pkg/build/webhook/generic/generic.go]: In the next version we really need to be using the encode decode libraries, with conversion, and add TypeMeta to the event.

Comment (smarterclayton) [inline pkg/build/webhook/generic/generic.go]: However we do want to allow web hooks to be empty (for ease of use), so the very strict rules api server follows don't apply.

**Comment by soltysh** [comment]
LGTM

**Comment by mfojtik** [comment]
LGTM

**Comment by csrwng** [comment]
@bparees @smarterclayton - merge ?

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**